File: backend/micro-feedback/src/main/java/com/example/elearning/entities/Feedback.java
package com.example.elearning.entities;

//import com.example.elearning.entities.FeedbackType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Feedback {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long feedbackId;

     // ID de l'utilisateur qui a soumis le feedback
    private String comments; // Commentaires textuels (si feedbackType = TEXT)
    private int rating; // Note donnée par l'utilisateur

    @Enumerated(EnumType.STRING)
    private FeedbackType feedbackType; // Type de feedback (TEXT ou AUDIO)

    private String audioFilePath; // Chemin du fichier audio (si feedbackType = AUDIO)
    private Date timestamp;
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false) // Clé étrangère vers la table User
    @JsonIgnore
    private User user;

    @Transient // Ce champ ne sera pas persistant en base de données
    private Long userId;// Utilisateur associé à ce feedback
}

File: frontend/integration/src/app/models/feedback.ts
export interface Feedback {
    feedbackId?: number; // Optionnel pour la création
    comments: string;
    rating: number;
    feedbackType: 'TEXT' | 'AUDIO';
    audioFilePath?: string; // Optionnel si feedbackType est TEXT
    userId: number;
    user?: {
        name: string;
      };
      timestamp?: string; // Add this if your backend returns it // ID de l'utilisateur associé
  }
  export interface PaginatedResponse<T> {
    content: T[];
    pageable: {
      pageNumber: number;
      pageSize: number;
    };
    totalElements: number;
    totalPages: number;
    last: boolean;
  }